tests/genomics_tests/test_genomic_cvl_pipeline.py

```diff
@@ -144,6 +148,120 @@ def test_w2sc_manifest_to_raw_ingestion(self):
+                {
+                    'decision_value': 'yes',
+                    'event_authored_time': datetime.datetime(2020, 10, 20)
+                },
+                {
+                    'decision_value': 'no',
+                    'event_authored_time': datetime.datetime(2020, 11, 17)
+                }
+            ]
+        )
+        # Member already in a the CVL workflow shouldn't be in a W1IL
+        self._generate_cvl_participant(
+            set_member_params={'gcSiteId': 'bcm'},
+            result_workflow_state={}  # letting defaults get set for the workflow state
+        )
+
+        manifest_generation_datetime = datetime.datetime(2021, 2, 7, 1, 13)
+        manifest_file_timestamp_str = manifest_generation_datetime.strftime("%Y-%m-%d-%H-%M-%S")
+
+        co_w1il_manifest = mock.MagicMock()
+        bcm_pgx_w1il_manifest = mock.MagicMock()
+        bcm_hdr_w1il_manifest = mock.MagicMock()
+
+        def side_effect(file_name):
+            if file_name == f'W1IL_manifests/BCM_AoU_CVL_W1IL_PGX_{manifest_file_timestamp_str}.csv':
+                return bcm_pgx_w1il_manifest
+            elif file_name == f'W1IL_manifests/BCM_AoU_CVL_W1IL_HDR_{manifest_file_timestamp_str}.csv':
+                return bcm_hdr_w1il_manifest
+            elif file_name == f'W1IL_manifests/CO_AoU_CVL_W1IL_PGX_{manifest_file_timestamp_str}.csv':
+                return co_w1il_manifest
+            else:
+                self.fail(f'Unexpected manifest generated: "{file_name}"')
+
+        exporter_instance = sql_exporter_class_mock.return_value
+        exporter_instance.open_cloud_writer.side_effect = side_effect
+
+        with clock.FakeClock(manifest_generation_datetime):
+            genomic_pipeline.cvl_w1il_manifest_workflow({
+                cvl_site_id: f'{cvl_site_id}_test_bucket'
+                for cvl_site_id in config.GENOMIC_CVL_SITES
+            })
+
+        self.assert_manifest_has_rows(
+            manifest_cloud_writer_mock=bcm_pgx_w1il_manifest,
+            expected_rows=[
+                self.expected_w1il_row(default_set_member, default_validation_metrics, default_summary),
+                self.expected_w1il_row(ny_set_member, ny_validation_metrics, ny_summary),
+                self.expected_w1il_row(male_set_member, male_validation_metrics, male_summary),
+                self.expected_w1il_row(hdr_and_pgx_set_member, hdr_and_pgx_validation_metrics, hdr_and_pgx_summary)
+            ]
+        )
+        self.assert_manifest_has_rows(
+            manifest_cloud_writer_mock=bcm_hdr_w1il_manifest,
+            expected_rows=[
+                self.expected_w1il_row(hdr_and_pgx_set_member, hdr_and_pgx_validation_metrics, hdr_and_pgx_summary)
+            ]
+        )
+        self.assert_manifest_has_rows(
+            manifest_cloud_writer_mock=co_w1il_manifest,
+            expected_rows=[
+                self.expected_w1il_row(co_set_member, co_validation_metrics, co_summary)
+            ]
+        )
+
```

Comment:
Nice use of mocks to test the row values. Can we also test the following? 
- The column headers in the CSV match what is expected by the CVLs. You may be doing this and I'm missing it.
- The `genomic_set_member` records are updated with the correct values for the HDR and PGx job run IDs. 
Note that the new job run IDs need to be added to `GenomicSetMemberDao.valid_job_id_fields` for this to work.
- That subsequent runs of the jobs don't include samples already sent in the W1IL for that module.
- The CSV files are being pushed to the correct bucket, i.e. Color isn't going to Broad's data bucket. This test is probably redundant, but it ensures the job is looking at the correct bucket-CVL mapping.

Reply:
> The column headers in the CSV match what is expected by the CVLs. You may be doing this and I'm missing it.

You're right, that's not part of the test right now. I'll add that in.

> The genomic_set_member records are updated with the correct values for the HDR and PGx job run IDs.
Note that the new job run IDs need to be added to GenomicSetMemberDao.valid_job_id_fields for this to work.

I will look into that, and may have other questions later. It seems like tasks are generated to do that and wouldn't actually execute while the test is running, but I might be able to figure something out. We may also need to add some of the other CVL jobs too, but I may be missing how they function.

> That subsequent runs of the jobs don't include samples already sent in the W1IL for that module.

Good catch, thanks. I thought I had one, but it looks like I modified it at some point. I'll add one in.

> The CSV files are being pushed to the correct bucket, i.e. Color isn't going to Broad's data bucket. This test is probably redundant, but it ensures the job is looking at the correct bucket-CVL mapping.

Will do 👍

Reply:
Thanks. I know a lot of this would have been clearer if we'd given you a tech design. for the job ID fields on the set members, you correctly added the field string to the manifest configs in the definition provider, i.e. `'job_run_field': 'cvlW1ilPgxJobRunId',`, but that string value needs to be added to the list here or else it won't actually write to the tables:`GenomicSetMemberDao.valid_job_id_fields`.

Reply:
`/rdr_service/dao/genomics_dao.py:222`. It looks like other CVL jobs need to be added to it too. I probably missed these in other reviews.

Reply:
The ManifestCompiler is set up to deploy a task that updates the job run ids on the GenomicSetMember. @j-kanuch, what do you think of having the test check that the task was created for setting the HDR and PGx job run ids?

Reply:
That should cover it since the job run field validation was removed in #2919.